include/zephyr/dt-bindings/clock/silabs/xg22-clock.h

```diff
@@ -0,0 +1,127 @@
+#define _CMU_CLKEN0_LDMAXBAR_BIT                 1
+#define _CMU_CLKEN0_RADIOAES_BIT                 2
+#define _CMU_CLKEN0_GPCRC_BIT                    3
+#define _CMU_CLKEN0_TIMER0_BIT                   4
+#define _CMU_CLKEN0_TIMER1_BIT                   5
+#define _CMU_CLKEN0_TIMER2_BIT                   6
+#define _CMU_CLKEN0_TIMER3_BIT                   7
+#define _CMU_CLKEN0_USART0_BIT                   8
+#define _CMU_CLKEN0_USART1_BIT                   9
+#define _CMU_CLKEN0_IADC0_BIT                    10
+#define _CMU_CLKEN0_AMUXCP0_BIT                  11
+#define _CMU_CLKEN0_LETIMER0_BIT                 12
+#define _CMU_CLKEN0_WDOG0_BIT                    13
+#define _CMU_CLKEN0_I2C0_BIT                     14
+#define _CMU_CLKEN0_I2C1_BIT                     15
+#define _CMU_CLKEN0_SYSCFG_BIT                   16
+#define _CMU_CLKEN0_DPLL0_BIT                    17
+#define _CMU_CLKEN0_HFRCO0_BIT                   18
+#define _CMU_CLKEN0_HFXO0_BIT                    19
+#define _CMU_CLKEN0_FSRCO_BIT                    20
+#define _CMU_CLKEN0_LFRCO_BIT                    21
+#define _CMU_CLKEN0_LFXO_BIT                     22
+#define _CMU_CLKEN0_ULFRCO_BIT                   23
+#define _CMU_CLKEN0_EUART0_BIT                   24
+#define _CMU_CLKEN0_PDM_BIT                      25
+#define _CMU_CLKEN0_GPIO_BIT                     26
+#define _CMU_CLKEN0_PRS_BIT                      27
+#define _CMU_CLKEN0_BURAM_BIT                    28
+#define _CMU_CLKEN0_BURTC_BIT                    29
+#define _CMU_CLKEN0_RTCC_BIT                     30
+#define _CMU_CLKEN0_DCDC_BIT                     31
+#define _CMU_CLKEN1_AGC_BIT                      0
+#define _CMU_CLKEN1_MODEM_BIT                    1
+#define _CMU_CLKEN1_RFCRC_BIT                    2
+#define _CMU_CLKEN1_FRC_BIT                      3
+#define _CMU_CLKEN1_PROTIMER_BIT                 4
+#define _CMU_CLKEN1_RAC_BIT                      5
+#define _CMU_CLKEN1_SYNTH_BIT                    6
+#define _CMU_CLKEN1_RDSCRATCHPAD_BIT             7
+#define _CMU_CLKEN1_RDMAILBOX0_BIT               8
+#define _CMU_CLKEN1_RDMAILBOX1_BIT               9
+#define _CMU_CLKEN1_PRORTC_BIT                   10
+#define _CMU_CLKEN1_BUFC_BIT                     11
+#define _CMU_CLKEN1_IFADCDEBUG_BIT               12
+#define _CMU_CLKEN1_CRYPTOACC_BIT                13
+#define _CMU_CLKEN1_RFSENSE_BIT                  14
+#define _CMU_CLKEN1_SMU_BIT                      15
+#define _CMU_CLKEN1_ICACHE0_BIT                  16
+#define _CMU_CLKEN1_MSC_BIT                      17
+#define _CMU_CLKEN1_TIMER4_BIT                   18
+
+/*
+ * DT macros for clock tree nodes.
+ * Defined as:
+ *  0..5 - Bit within CLKEN register
+ *  6..8 - CLKEN register number
+ * Must stay in sync with equivalent SL_BUS_*_VALUE constants in the Silicon Labs HAL to be
+ * interpreted correctly by the clock control driver.
+ */
+#define CLOCK_AGC               ((BUS_CLOCK_CLKEN1 << 6) | _CMU_CLKEN1_AGC_BIT)
```

Comment:
I clean solution would be use `#size-cells = 3`. I assume Zephyr does not support it?

I believe `_CMU_CLKENx_xxx_BIT` is a useless layer of indirection.
BTW, `<< 6` is redundant with `CLOCK_REG_MASK`. I suggest:
```
#define CLOCK_AGC     (FIELD_PREP(CLOCK_REG_MASK, 1) | FIELD_PREP(CLK_REG_MASK, 11))
```

Reply:
We could increase size-cells to 3, but then the headers effectively become pointless, and we might as well put the pure integers straight into DT. I would prefer using the headers to create easy-to-read values for use in DT.

As for FIELD_PREP, I don't disagree, but then we need to move it from util.h to util_macro.h, since util.h isn't available from DT.

Reply:
Pulled in FIELD_PREP and used it to build the bitfields.